src/utils/session.ts

```diff
@@ -1,24 +1,46 @@
+import { defaultDetails } from "../constants";
 import { Request } from "express";
-import { isDetails } from "private-api-sdk-node/dist/services/presenter-account/types";
-import { type Details } from "private-api-sdk-node/src/services/presenter-account/types";
+import { Name, type Details } from "private-api-sdk-node/src/services/presenter-account/types";
 
 export const PRESENTER_ACCOUNT_SESSION_KEY = "presenter_account_details";
 
 export function getPresenterAccountDetails(req: Request): Details {
-    const presenterAccountDetails = req.session?.getExtraData(PRESENTER_ACCOUNT_SESSION_KEY);
-
-    if (presenterAccountDetails === undefined) {
-        throw new Error(`Presenter account details are undefined`);
+    let presenterAccountDetails: Details | undefined;
+    if (req.session !== undefined){
+        presenterAccountDetails = req.session.getExtraData(PRESENTER_ACCOUNT_SESSION_KEY);
+        if (presenterAccountDetails === undefined) {
+            presenterAccountDetails =  fetchUserDetails(req, defaultDetails);
+            setPresenterAccountDetails(req, defaultDetails);
+        }
+        return presenterAccountDetails;
     }
+    return fetchUserDetails(req, defaultDetails);
+}
 
-    if (!isDetails(presenterAccountDetails)) {
-        throw new Error(`Invalid presenter account details`);
-    }
 
-    return presenterAccountDetails;
+export function setPresenterAccountDetails(req: Request, details: Details) {
+    if (req.session !== undefined){
+        req.session.setExtraData(PRESENTER_ACCOUNT_SESSION_KEY, details);
+    }
```

Comment:
This change just makes it more verbose. It should use nullish coalescing to make it clearer.
```suggestion
    req.session?.setExtraData(PRESENTER_ACCOUNT_SESSION_KEY, details);
 
```